### PR TITLE
Switch to a YAML configuration, and handle/convert old TOML config too

### PR DESCRIPTION
**Pull request by j0ni**

### CODE REVIEW

File: conf/common.go [hunk 1]
```diff
@@ -46,99 +44,101 @@ func fileExists(fname string) bool {
...
-		log.Error(err)
-	}
-
-	return newPath
-}
-
 func convertOldConf() (c *Conf) {
 	env := FetchEnv()
 	log := FetchLog()
+	var conf_file, var_file string
```
Comment (j0ni) [inline conf/common.go]: lol you really don't like other languages do you?

File: main.go [hunk 2]
```diff
@@ -140,7 +144,7 @@ func initialize(env *agent.Env) *agent.Agent {
 			// exponential decay library; by the time we hit this
 			// point we've been trying for about, what, an hour?
 			log.Infof("Register server error: %s", err)
-			err = a.RegisterServer()
+			err = a.RegisterServer() // WTF
```
Comment (j0ni) [inline main.go]: drive by, no idea. I removed it

File: conf/env.go [hunk 3]
```diff
@@ -94,11 +94,15 @@ func InitLogging() {
 	// TODO: SetLevel must come before SetBackend
 	format := logging.MustStringFormatter("%{time} %{pid} %{shortfile}] %{message}")
 	stdoutBackend := logging.NewBackendFormatter(logging.NewLogBackend(os.Stdout, "", 0), format)
-	var err error
 	if env.Prod {
 		logging.SetLevel(logging.INFO, "canary-agent")
 
-		conf := NewConfFromEnv()
+		conf, err := NewConfFromEnv()
```
Comment (j0ni) [inline conf/env.go]: Not sure I understand the question.

File: conf/common.go (hunk 1 above)
Comment (phillmv) [inline conf/common.go]: d'oh! this was not intentional :(.

File: conf/env.go (hunk 3 above)
Comment (phillmv) [inline conf/env.go]: I think it made sense before my own edits; time to disregard.

**Pull request merged**